core/templates/services/context.service.ts

```diff
@@ -251,10 +256,23 @@ export class ContextService {
       ServicesConstants.PAGE_CONTEXT.COLLECTION_EDITOR,
       ServicesConstants.PAGE_CONTEXT.TOPIC_EDITOR,
       ServicesConstants.PAGE_CONTEXT.STORY_EDITOR,
-      ServicesConstants.PAGE_CONTEXT.SKILL_EDITOR
+      ServicesConstants.PAGE_CONTEXT.SKILL_EDITOR,
+      ServicesConstants.PAGE_CONTEXT.TOPICS_AND_SKILLS_DASHBOARD
     ];
     return (allowedPageContext.includes(currentPageContext));
   }
+
+  setSaveImagesToLocalStorageContext(): void {
```

Comment:
Hm, I don't really understand this concept. What is a "save images to local storage" context? In what way is it a context? Why do we even need a context -- why can't we just save them to local storage directly?

Reply:
This sets the context for the directives that handle image storage. So, in those directives, based on whether this context is set or not, will save images locally or upload them.
It basically acts as a flag for the image directives.

Reply:
Then the naming is quite confusing. I think you'll want a two-valued flag there that is something like `imageSaveDestination =  IMAGE_SAVE_DESTINATION_SERVER;`, and then you have methods to change it to IMAGE_SAVE_DESTINATION_LOCAL_STORAGE and back again.

The "set" method would be: setImageSaveDestination(...) with the flag passed in. Make sure that the values passed in are named constants.

Reply:
Done